tests/expectations/core/test_expect_column_values_to_be_present_in_other_table.py

```diff
@@ -0,0 +1,156 @@
+        foreign_key_column="CUSTOMER_ID",
+        foreign_table="customer_table",
+        foreign_table_key_column="CUSTOMER_ID",
+    )
+    assert res.success is True
+
+
+def test_failed_expectation_run(referential_integrity_db):
+    context = gx.get_context(cloud_mode=False)
+    datasource_name = "my_snowflake_datasource"
+    datasource = context.sources.add_sqlite(
+        datasource_name,
+        connection_string="sqlite:///../../test_sets/referential_integrity_dataset.db",
+    )
+    asset_name = "order_table_2"
+    asset = datasource.add_table_asset(name=asset_name, table_name="order_table_2")
+    batch_request = asset.build_batch_request()
+    expectation_suite_name = "my_suite"
+    context.add_or_update_expectation_suite(
+        expectation_suite_name=expectation_suite_name
+    )
+    validator = context.get_validator(
+        batch_request=batch_request,
+        expectation_suite_name=expectation_suite_name,
+    )
+    res = validator.expect_column_values_to_be_present_in_another_table(
+        foreign_key_column="CUSTOMER_ID",
+        foreign_table="customer_table",
+        foreign_table_key_column="CUSTOMER_ID",
+    )
+    assert res.success is False
+    assert res["result"]["observed_value"] == "2 missing values."
+    assert res["result"]["unexpected_index_list"] == [
+        {"CUSTOMER_ID": 5},
+        {"CUSTOMER_ID": 6},
+    ]
+
+
+def test_invalid_configurations(referential_integrity_db):
+    context = gx.get_context(cloud_mode=False)
+    datasource_name = "my_snowflake_datasource"
+    datasource = context.sources.add_sqlite(
+        datasource_name,
+        connection_string="sqlite:///../../test_sets/referential_integrity_dataset.db",
+    )
+    asset_name = "order_table_2"
+    asset = datasource.add_table_asset(name=asset_name, table_name="order_table_2")
+    batch_request = asset.build_batch_request()
+    expectation_suite_name = "my_suite"
+    context.add_or_update_expectation_suite(
+        expectation_suite_name=expectation_suite_name
+    )
+    validator = context.get_validator(
+        batch_request=batch_request,
+        expectation_suite_name=expectation_suite_name,
+    )
+    with pytest.raises(KeyError):
+        # missing keys
+        validator.expect_column_values_to_be_present_in_another_table(
+            foreign_table_key_column="CUSTOMER_ID",
+        )
+
+    with pytest.raises(gx_exceptions.MetricResolutionError):
+        # missing column
+        validator.expect_column_values_to_be_present_in_another_table(
+            foreign_key_column="I_DONT_EXIST",
+            foreign_table="customer_table",
+            foreign_table_key_column="CUSTOMER_ID",
+        )
+
+    with pytest.raises(KeyError):
+        # invalid parameter
+        validator.expect_column_values_to_be_present_in_another_table(
+            i_dont_exist="CUSTOMER_ID",
+            foreign_table="customer_table",
+            foreign_table_key_column="CUSTOMER_ID",
+        )
```

Comment:
Nit - not a requirement but could optionally parameterize the test or create three separate ones to keep things tightly scoped. Not a blocker

Reply:
Same note about validator's dynamic method

Reply:
Actually the more I think about this, I don't think we need to even test invalid params - all Pydantic models fail in the same manner when this occurs

Reply:
I elected to split it up. The unittest you suggested actually tests two of the cases in a simpler way, and the one where the missing `I_DONT_EXIST` column is given, the `batch.validate()` actually swallows the Exception and doesn't raise. 

This is what I was imagining would happen: 
```python
checkpoint_result = checkpoint.run(
    runtime_configuration={
        "catch_exceptions": False,
    }
)
```

But, I was able to do the same test by checking the result's `exception_message` which the same test.